ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -432,9 +434,11 @@ private String formatExpression(final Expression expression) {
       final String instanceName = funNameToCodeName.apply(functionName);
 
       final UdfFactory udfFactory = functionRegistry.getUdfFactory(node.getName());
-      final List<SqlType> argumentSchemas = node.getArguments().stream()
-          .map(expressionTypeManager::getExpressionSqlType)
-          .collect(Collectors.toList());
+      final List<SqlArgument> argumentSchemas = new ArrayList<>();
+      for (final Expression argExpr : node.getArguments()) {
+        final SqlType newSqlType = expressionTypeManager.getExpressionSqlType(argExpr);
```

Comment:
This is my bad, I missed this when doing some updates. We should be able to leave the original implementation and just add `map(arg -> sqlArgument.of(arg.getSqlType()))`